lib/SILOptimizer/Mandatory/AddressLowering.cpp

```diff
@@ -1712,11 +1763,41 @@ AddressMaterialization::materializeProjectionIntoUse(Operand *operand,
       return pass.function->getArguments()[resultIdx];
     }
     SILValue tupleAddr = materializeComposingUser(tupleInst, intoPhiOperand);
-    return B.createTupleElementAddr(pass.genLoc(), tupleAddr,
-                                    operand->getOperandNumber(),
-                                    operand->get()->getType().getAddressType());
+    return projectionBuilder.createTupleElementAddr(
+        pass.genLoc(), tupleAddr, operand->getOperandNumber(),
+        operand->get()->getType().getAddressType());
+  }
+  }
+}
+
+SILInstruction *
+AddressLoweringState::getProjectionInsertionPoint(SILValue userValue) const {
+  if (!valueStorageMap.contains(userValue)) {
+    return userValue->getDefiningInsertionPoint();
+  }
+  if (!valueStorageMap.getStorage(userValue).isProjection()) {
+    return userValue->getDefiningInsertionPoint();
```

Comment:
Hopefully these two special cases go away if moved inside AddressMaterialization. If not, they could be folded into a single helper.

Reply:
Replaced both with defaulting `retval = userValue->getDefiningInsertionPoint();` before the loop.